From 1f01415ba33d58185ec80762f5eaa9451597c292 Mon Sep 17 00:00:00 2001
From: Guillermo Molteni <gmolteni@fusap.com.ar>
Date: Mon, 26 Nov 2018 08:48:14 +0100
Subject: Don't reverse order of files opened from the command line

Summary:
When you do kate file1 file2 file3, it will open tabs
in same order, file1 file2 file3.
In case of having kate opened, invoking it again with
kate file4 file5 file6 final result on tabs shall be
file4 file5 file6 file1 file2 file3
BUG: 397913

Reviewers: cullmann

Reviewed By: cullmann

Subscribers: zetazeta, cullmann, brauch, kwrite-devel

Tags: #kate

Differential Revision: https://phabricator.kde.org/D17145
---
 kate/kateapp.cpp | 5 ++++-
 kate/main.cpp    | 4 +++-
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/kate/kateapp.cpp b/kate/kateapp.cpp
index 174613e..90cdd15 100644
--- a/kate/kateapp.cpp
+++ b/kate/kateapp.cpp
@@ -175,7 +175,10 @@ bool KateApp::startupKate()
     KTextEditor::Document *doc = nullptr;
     const QString codec_name = codec ? QString::fromLatin1(codec->name()) : QString();
 
-    Q_FOREACH(const QString positionalArgument, m_args.positionalArguments()) {
+    //  Bug 397913: Reverse the order here so the new tabs are opened in same order as the files were passed in on the command line
+    QString positionalArgument;
+    for (int i = m_args.positionalArguments().count() - 1; i >= 0; --i) {
+        positionalArgument = m_args.positionalArguments().at(i);
         UrlInfo info(positionalArgument);
 
         // this file is no local dir, open it, else warn
diff --git a/kate/main.cpp b/kate/main.cpp
index c1ae2c5..a45f0d7 100644
--- a/kate/main.cpp
+++ b/kate/main.cpp
@@ -416,7 +416,9 @@ int main(int argc, char **argv)
             QStringList tokens;
 
             // open given files...
-            foreach(const QString & url, urls) {
+            // Bug 397913: Reverse the order here so the new tabs are opened in same order as the files were passed in on the command line
+            for (int i = urls.size() - 1; i >= 0; --i) {
+                const QString &url = urls[i];
                 QDBusMessage m = QDBusMessage::createMethodCall(serviceName,
                                 QStringLiteral("/MainApplication"), QStringLiteral("org.kde.Kate.Application"), QStringLiteral("tokenOpenUrlAt"));
 
-- 
cgit v1.1

